sql/moz-fx-data-shared-prod/marketing_suppression_list_external/campaign_monitor_suppression_list_v1/query.py

```diff
@@ -0,0 +1,137 @@
+"""Suppression List downloaded from CampaignMonitor API and uploaded to BigQuery."""
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, List
+
+import requests
+import rich_click as click
+import yaml
+from google.cloud import bigquery
+from requests.auth import HTTPBasicAuth
+
+DEFAULT_PROJECT_ID = Path(__file__).parent.parent.parent.name
+DEFAULT_DATASET_ID = Path(__file__).parent.parent.name
+DEFAULT_TABLE_NAME = Path(__file__).parent.name
+
+BASE_URL = "https://api.createsend.com/api/v3.3/clients"
+SCHEMA_FILE = Path(__file__).parent / "schema.yaml"
+SCHEMA = bigquery.SchemaField.from_api_repr(
+    {"name": "root", "type": "RECORD", **yaml.safe_load(SCHEMA_FILE.read_text())}
+).fields
+
+
+def get_api_response(url: str, api_key: str):
+    """Get json of response if the API request was successful."""
+    headers = {"Accept": "application/json"}
+    auth = HTTPBasicAuth(api_key, "")
+
+    response = requests.get(url=url, headers=headers, auth=auth)
```

Comment:
Is there a particular reason you opted not to use the [Python package provided by Campaign Monitor](https://github.com/campaignmonitor/createsend-python)? Its latest commit is a couple of years ago, so maybe because it's out of date with their API or incompatible with Python 3.11?

Reply:
https://github.com/campaignmonitor/createsend-python/blob/b7eb6a8eeb15233439c6e4ba0406481f28fca209/setup.py#L52
They only support till python 3.10 according to this
It's been 3 years since they updated 
and https://github.com/campaignmonitor/createsend-python/blob/b7eb6a8eeb15233439c6e4ba0406481f28fca209/lib/createsend/client.py#L77 looking at the suppression list code I would still need to take care of calling it multiple times myself ... 
so this just felt ... more straight forward